lib: gui: lvgl: align names and other improvements

gmarull opened the pull request:
Summary:

- Kconfig object names have been aligned with LVGL (as discussed and agreed in dev-review)
- Improved lv_conf.h by introducing `IS_ENABLED` macro for boolean options
- Splitted Kconfig options into more granular units
- Sorted objects in order to improve readability/keep track of options

gmarull committed:
lib: gui: lvgl: align all Kconfig object names

Align all Kconfig option names with LVGL names. The followed rule:
LV_(.*) -> CONFIG_LVGL_(.*).

Also replaced LVGL boolean configuration entries using if/else/endif
with direct IS_ENABLED macro.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

pabigot reviewed (commented):
A request for the commit "lib: gui: lvgl: use IS_ENABLED for boolean options"

Please remove the @ before my github handle in the commit message.  Every time you push an update and the SHA changes I get an email.  (I'm fine with you dropping the NOTE entirely; I appreciate the thought but it's not like the idea has major value.)

vanwinkeljan commented on lib/gui/lvgl/Kconfig.graphical:
```diff
@@ -0,0 +1,139 @@
+	help
+	  Name of the display device to use for rendering.
+
+config LVGL_HOR_RES
```
Should we not also align `LVGL_HOR_RES` and `LVGL_VER_RES ` with LVGL defines  `LV_HOR_RES_MAX` and `LV_VER_RES_MAX`?

gmarull replied:
@vanwinkeljan There are some more cases like this one (e.g. `LV_INDEV_DEF_READ_PERIOD` uses `CONFIG_LVGL_INPUT_REFRESH_PERIOD`). Should we do a complete alignment then, including stuff like `CONFIG_LVGL_USE_LED` for `LV_USE_LED` instead of `CONFIG_LVGL_OBJ_LED`? Initial idea was to just align object names.

vanwinkeljan replied:
I would line up all of them instead of having a mix

gmarull replied:
@vanwinkeljan I have adjusted all names now (I hope I haven't missed any). Can you re-review now? I'll split files once name changes are approved.

vanwinkeljan replied:
@gmarull could it be you missed `Kconfig.graphical`?

gmarull replied:
Added missing suffix _MAX for LVGL_HOR_RES and LVGL_VER_RES

vanwinkeljan commented on lib/gui/lvgl/Kconfig.graphical:
```diff
@@ -0,0 +1,139 @@
+
+config LVGL_HOR_RES
+	int "Horizontal Screen Resolution"
+	default 320
```
Is this an intentional change (same for vertical res and dpi) from LVGL defaults?

gmarull replied:
Ooops, good catch. I re-used last PR initial commit and missed that one.

gmarull committed:
docs: releases: 2.4: add notes about LVGL Kconfig changes

Add notes about LVGL Kconfig options name changed after alignment
with upstream names.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull committed:
lib: gui: lvgl: split Kconfig settings

LVGL Kconfig settings have been splitted into more granular units in
order to improve readability and maintenance.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull committed:
lib: gui: lvgl: sort objects

LVGL objects have been alphabetically sorted. The aim of this change is
to help locating objects as well as keeping track of them.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull committed:
lib: gui: lvgl: improve cmake formatting

Improve the format of the CMakeLists.txt file.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull commented:
@vanwinkeljan should be ready now
